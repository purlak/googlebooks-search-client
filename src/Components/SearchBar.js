import React, { Component } from 'react';
import BookService from '../Services/BookService.js';
import '../css/App.css';

class SearchBar extends Component {
  constructor (props) {
    super (props);
    this.state = {
      searchTerm: '',
      books: this.props.books,
    };
  }
  handleSearchInput = event => {
    this.setState({searchTerm: event.target.value });
  }

  searchBook = () =>  {
    BookService.fetchBooks(this.state.searchTerm)
      .then(data => this.props.onBooksChange(data.items))
  }

  render () {
    return (
      <div>
        <input className="searchBar"
          type="text"
          onChange={this.handleSearchInput}
          onKeyPress={event => {
            if (event.key === 'Enter') {
              this.searchBook()
            }
          }}
          value={this.state.searchTerm}
          placeholder="search books"
        />
        <button className="buttonSize" onClick={this.searchBook}>Search</button>
      </div>
  )}
}

export default SearchBar;
