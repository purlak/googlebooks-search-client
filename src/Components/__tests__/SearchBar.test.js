import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from '../SearchBar';

describe ("SearchBar component", () => {
    test ("renders", () => {
        const wrapper = shallow(<SearchBar />);
        expect(wrapper.exists()).toBe(true);
    });

    test ("search text is echoed", () => {
      const wrapper = shallow (<SearchBar searchBook={() => {}} />);
      wrapper.find("input").simulate("change", {
        target: { value: "hello"  }
      });
      expect(wrapper.find("input").props().value).toEqual("hello");
    });

    test('pass value to the onChange handler', () => {
    const searchTerm = 'hello';
    const onChange = jest.fn();
    const wrapper = shallow(
        <SearchBar onChange={searchBook} />
    );

    expect(wrapper).toMatchSnapshot();
        wrapper.find('button').simulate('click', {
        target: { searchBook },
    });

    expect(onChange).toBeCalledWith(searchBook);
  });
})
