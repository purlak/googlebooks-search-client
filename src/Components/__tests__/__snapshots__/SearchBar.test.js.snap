// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component pass value to the onChange handler 1`] = `
<div>
  <input
    className="searchBar"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="search books"
    type="text"
    value=""
  />
  <button
    className="buttonSize"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
